uint64 time
int32 ranking
int32 score